chain/x/shard/keeper/query_server.go

```diff
@@ -2,39 +2,40 @@ package keeper
 
 import (
 	"context"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	"github.com/argus-labs/world-engine/chain/x/shard/types"
 )
 
 var _ types.QueryServer = &Keeper{}
 
-func (k *Keeper) Batches(ctx context.Context, req *types.QueryBatchesRequest) (*types.QueryBatchesResponse, error) {
+func (k *Keeper) Transactions(ctx context.Context, req *types.QueryTransactionsRequest,
+) (*types.QueryTransactionsResponse, error) {
 	if req.Namespace == "" {
 		return nil, sdkerrors.ErrInvalidRequest.Wrap("namespace required but not supplied")
 	}
 	key, limit := types.ExtractPageRequest(req.Page)
-	res := types.QueryBatchesResponse{
-		Batches: make([]*types.TransactionBatch, 0, limit),
-		Page:    &types.PageResponse{},
+	res := types.QueryTransactionsResponse{
+		Txs:  make([]*types.TickedTransactions, 0, limit),
+		Page: &types.PageResponse{},
 	}
 	count := uint32(0)
-	k.iterateBatches(sdk.UnwrapSDKContext(ctx), key, nil, req.Namespace, func(tick uint64, batch []byte) bool {
-		// we keep the check here so that if we hit the limit,
-		// we return the NEXT key in the iteration, not the one before it.
-		if count == limit {
-			res.Page.Key = k.bytesForUint(tick)
-			return false
-		}
-		res.Batches = append(res.Batches, &types.TransactionBatch{
-			Namespace: req.Namespace,
-			Tick:      tick,
-			Batch:     batch,
+	k.iterateTransactions(sdk.UnwrapSDKContext(ctx), key, nil,
+		req.Namespace, func(tick uint64, txs *types.Transactions) bool {
+			// we keep the check here so that if we hit the limit,
+			// we return the NEXT key in the iteration, not the one before it.
+			if count == limit {
+				res.Page.Key = k.getTransactionKey(tick)
+				return false
+			}
+			res.Txs = append(res.Txs, &types.TickedTransactions{
+				Tick: tick,
```

Comment:
i would generalize this to "epoch". in the future, there might be other shards that doesn't have the concept of tick.

Reply:
[refactor: tick -> epoch](https://github.com/Argus-Labs/world-engine/pull/169/commits/4c17cdd25f6260b3f3d5752495fbc08a4de76392) done